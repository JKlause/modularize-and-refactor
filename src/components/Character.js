import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ img, name, species }) => (
  <dl>
    <dt>Img</dt>
    <dd>{img}</dd>
  
    <dt>Name</dt>
    <dd>{name}</dd>
  
    <dt>species</dt>
    <dd>{species}</dd>
  </dl>
);

export default Character;

Character.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.number.isRequired,
  species: PropTypes.string.isRequired
};
