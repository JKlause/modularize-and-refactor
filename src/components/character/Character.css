.character {
  margin: 10px;
  padding: 10px;
  background-color: grey;
  border: 4px solid black;
  text-align: center;
  img {
    width: 250px;
    border: 2px solid; 
  }
  h2, h3 {
    border-bottom: 2px solid black
  }
}