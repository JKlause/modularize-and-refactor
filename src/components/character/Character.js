import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ img, name, species }) => (
  <div>
    <img src={img} />
    <h2>Name: {name}</h2>
    <h3>Species: {species}</h3>
  </div>
);

export default Character;

Character.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired
};
