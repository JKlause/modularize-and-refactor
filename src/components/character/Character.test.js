import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', ()=> {
  it('matches a snapshot', () => {
    const character = shallow(<Character 
      img='https://rickandmortyapi.com/api/character/avatar/2.jpeg'
      name='Morty Smith'
      species='Human'
    />);
    expect(character).toMatchSnapshot();
  });
});
