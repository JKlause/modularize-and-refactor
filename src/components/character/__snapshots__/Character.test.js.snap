// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component matches a snapshot 1`] = `
<div>
  <img
    src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
  />
  <h2>
    Name: 
    Morty Smith
  </h2>
  <h3>
    Species: 
    Human
  </h3>
</div>
`;
